clang/lib/Sema/SemaExpr.cpp

```diff
@@ -12186,13 +12186,39 @@ Sema::CheckValueTerminatedAssignmentConstraints(QualType LHSType,
              Context.hasSameUnqualifiedType(
                  Context.getCorrespondingSignedType(LPointee),
                  Context.getCorrespondingSignedType(RPointee)));
-        if (CompatiblePointees &&
+        if (CompatiblePointees && RHSExpr->isPRValue() &&
             RHSExpr->tryEvaluateTerminatorElement(Evald, Context) &&
             Evald.Val.isInt() &&
             llvm::APSInt::isSameValue(LVTT->getTerminatorValue(Context),
                                       Evald.Val.getInt())) {
           return Sema::Compatible;
         }
+        // If in C++ Safe Buffers/Bounds Safety interoperation mode, the RHS can
+        // be 'std::string::c_str':
+        bool SafeBuffersEnabled = !Diags.isIgnored(
+            diag::warn_unsafe_assign_to_null_terminated_pointer,
+            RHSExpr->getBeginLoc());
+
+        if (LangOpts.CPlusPlus && SafeBuffersEnabled) {
+          if (const auto *MCE =
+                  dyn_cast<CXXMemberCallExpr>(RHSExpr->IgnoreParenImpCasts())) {
+            IdentifierInfo *MethodDeclII = nullptr, *ObjTypeII = nullptr;
```

Comment:
I think the variable names could be better here.   MethodDeclI1 and ObjTypeI1 are hard tor ead.

Reply:
They are not `MethodDeclI1` or `ObjTypeI1`.  They are two capital "i"s.  I saw some other places in the code base where variables holding `IdentifierInfo` are named `II`, so I just followed the convention.